src/widgets/SearchBar.js

```diff
@@ -37,30 +40,49 @@ export const SearchBar = ({
   debounceTimeout,
   ...textInputProps
 }) => {
+  const [textValue, setTextValue] = useState('');
+
+  // Set the internal text value if it differs from the passed
+  // value. Keeping the search bar text input up to date with
+  // user input while debouncing actual filtering of data.
+  useEffect(() => {
+    if (value !== textValue) setTextValue(value);
+  }, [value]);
+
   const internalViewStyle = { ...viewStyle, borderColor: color };
   const internalTextStyle = { ...textInputStyle, color };
 
-  const debouncedCallback = useMemo(
-    () => newValue => debounce(onChangeText(newValue), debounceTimeout),
-    [onChangeText]
+  // For Debouncing to funtion, need to use the same function as the
+  // last call. Memoize this to keep the reference semi-stable. Keeping
+  // this reference is an optimization, not a requirement so can change.
```

Comment:
I mean, there are no dependancies given to useCallBack so it will be stable and won't change if you try right?

Reply:
there's always a chance memoized data can be kicked out of the cache